bpf/flows.c

```diff
@@ -93,6 +93,7 @@ static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
     // a spinlocked alternative version and use it selectively https://lwn.net/Articles/779120/
     flow_metrics *aggregate_flow = (flow_metrics *)bpf_map_lookup_elem(&aggregated_flows, &id);
     if (aggregate_flow != NULL) {
+        bpf_spin_lock(&aggregate_flow->lock);
```

Comment:
we probably need to check the return code for spin lock and unlock and log an error

Reply:
actually, there's no error code if I'm correct. The doc for bpf_spin_lock says
```
 * Returns
 * 	0
```
In linux helper.c it returns void: https://elixir.bootlin.com/linux/v6.12.1/source/kernel/bpf/helpers.c#L283

Reply:
doc said it was returning something https://docs.ebpf.io/linux/helper-function/bpf_spin_lock/ maybe doc is wrong

Reply:
it does, but apparently it's always 0, not sure why : https://github.com/netobserv/netobserv-ebpf-agent/blob/main/bpf/headers/bpf_helper_defs.h#L2359-L2360